src/services/iiif-parser.js

```diff
@@ -0,0 +1,248 @@
+}
+
+function filtersrc(sources) {
+  if (sources.length < 1) {
+    throw 'Error fetching media files. Please check the Manifest.';
+  }
+  else if (sources.length == 1) {
+    return sources[0].id;
+  } else {
+    let srcId = sources[0].id;
+    sources.map((src) => {
+      const srcQuality = src.getLabel()[0].value.toLowerCase();
+      if (srcQuality == 'auto' || srcQuality == 'low') {
+        srcId = src.id;
+      }
+    });
+    return srcId;
+  }
+}
+
+
+/**
+ * Retrieve the list of alternative representation files in manifest or canvas
+ * level to make available to download
+ * @param {Object} manifest
+ * @param {Number} canvasIndex
+ * @returns List of files under `rendering` property in manifest
+ */
+export function getWaveformInfo(manifest, canvasIndex) {
+  let waveformFile = null;
+  let canvasRendering = [];
+
+  try {
+    const manifestParsed = parseManifest(manifest);
+
+    let canvas = manifestParsed.getSequences()[0]
+      .getCanvasByIndex(canvasIndex);
+    if(canvas.__jsonld.rendering) {
+      canvasRendering = canvas.__jsonld.rendering;
+
+      if (canvasRendering.length > 0) {
+        const w = canvasRendering[0];
+        const name = getLabelValue(w.label);
+        if (w.format == 'application/json' && name == 'waveform.json') {
+          waveformFile = w.id;
+        }
+      }
+    }
+  } catch (err) {
+    console.error(err);
+  }
+  return waveformFile;
+}
+
+/**
+ * Parse the structures within manifest into a nested JSON object
+ * structure to be consumed by the ReactJS components to help visualize
+ * and edit structure
+ * @param {Object} manifest current manifest
+ * @param {Number} duration duration of the canvas in manifest
+ * @param {Object} initStructure initial structure by default from props
+ * @returns {Array.<Object>} structureJSON - array of nested JSON 
+ * objects with structure items parsed from the given manifest
+ */
+export function parseStructureToJSON(manifest, initStructure, duration) {
+  let structureJSON = [];
+
+  let buildStructureItems = (items, children) => {
+    if (items.length > 0) {
+      items.map((i) => {
+        const childCanvases = getRangeCanvas(i.id, manifest);
+        let structItem = {};
+        if (childCanvases.length > 0) {
+          const { start, end } = getMediaFragment(childCanvases[0], duration);
+          structItem = {
+            label: getLabelValue(i.label),
+            type: "span",
+            begin: smUtils.toHHmmss(start),
+            end: smUtils.toHHmmss(end),
+          };
+          children.push(structItem);
+        } else {
+          structItem = {
+            label: getLabelValue(i.label),
+            type: "div",
+            items: [],
+          };
+          if (i.items) {
+            buildStructureItems(i.items, structItem.items);
+          }
+          children.push(structItem);
+        }
+      });
+    }
+  };
+
+  let manifestName;
+  let structures = [];
+
+  if (manifest != undefined || manifest != null) {
```

Comment:
Should this be `&&`?
```suggestion
  if (manifest != undefined && manifest != null) {
```

Reply:
Absolutely! Good catch :100: